react-components/jsx.md

```diff
@@ -106,52 +98,124 @@ In most contexts, we can wrap JavaScript expressions in parentheses `()`, which
 
 ### !end-callout
 
-### JSX Objects With Children: `App.js`
+### JSX Objects With Children: `App.jsx`
 
-Let's revisit this `src/App.js`:
+Let's revisit this `src/App.jsx`:
 
 ```js
 function App() {
+  const [count, setCount] = useState(0)
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
+    <>
+      <div>
+        <a href="https://vitejs.dev" target="_blank">
+          <img src={viteLogo} className="logo" alt="Vite logo" />
+        </a>
+        <a href="https://react.dev" target="_blank">
+          <img src={reactLogo} className="logo react" alt="React logo" />
+        </a>
+      </div>
+      <h1>Vite + React</h1>
+      <div className="card">
+        <button onClick={() => setCount((count) => count + 1)}>
+          count is {count}
+        </button>
         <p>
-          Edit <code>src/App.js</code> and save to reload.
+          Edit <code>src/App.jsx</code> and save to test HMR
         </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+      </div>
+      <p className="read-the-docs">
+        Click on the Vite and React logos to learn more
+      </p>
+    </>
+  )
 }
 ```
 
+<br />
+
+<details>
+<summary>HMR: Hot Module Reloading</summary>
+
+Hot Module Reloading, or HMR, as mentioned in `App.jsx` is a feature that the Vite development server comes with that automatically reloads the app when we edit the source code. 
+
+<br />
+
+This is a useful development feature, but because of how it can interact with editing files in VS Code, this is why we recommend turning off the VSCode Auto Save feature when working with a React project.
+
+</details>
+
+<br />
+
+### !callout-info
+
+## JSX Expressions Must Have One Parent Element!
+
+In JSX, there is a rule that states that a JSX object must always return a single element. This rule applies to React, which means that every component can only return a single root element.
+
+<br />
```

Comment:
I think a little more in the visible area woudl be helpful. How about something like:

```suggestion
In JSX, there is a rule that states that a JSX object must always return a single element. This rule applies to React, which means that every component can only return a single root element.

<br />

The Vite template meets this requirement by wrapping its JSX markup in a special [fragment](https://react.dev/reference/react/Fragment) tag. Keep reading for more about fragments, and alternative approaches we can use in our own components.

<br />

```

And then update the summary label and maybe do some minor edits to the hidden content?